packages/tracking_repository/lib/src/tracking_repository.dart

```diff
@@ -1,18 +1,16 @@
+import 'dart:developer' as dev;
+import 'package:firebase_analytics/firebase_analytics.dart';
 import 'package:tracking_repository/src/track_event.dart';
 
 /// Used to interface tracking services
 class TrackingRepository {
-  /// final Tracker? tracker;
-  /// Loads trackers
-  /// void configureWith(
-  ///   // firebase
-  ///   // GA
-  ///   // amplitude
-  ///   // mixpanel
-  ///   // ...
-  /// ) {dart analyze
-  ///    this.trakingService = trakingService
-  /// }
+  /// Default construcot
+  TrackingRepository({
+    required this.firebase,
+  });
```

Comment:
Please use the initializer list to initialize it like: `: _firebaseAnalytics = firebaseAnalytics;` since we don't wanna be able to access our dependencies directly.